app/components/cases/new-subcase.js

```diff
@@ -141,10 +141,14 @@ export default class CasesNewSubcase extends Component {
 
   @action
   async copySubcaseProperties(subcase, latestSubcase, fullCopy, pieces) {
+    const subcaseTypeWithoutMandatees = [
+      CONSTANTS.SUBCASE_TYPES.BEKRACHTIGING,
+    ].includes(subcase.type?.get('uri'));
```

Comment:
De `includes()` gaat niet wachten op het resolven van de promise. Ik vermoed dat dit nu in de praktijk werkt omdat de data al beschikbaar is, en de promise dus onmiddellijk resolved met de URI ,maar hier kunnen we niet vanuit gaan.
We kunnen hier toch expliciet `await`-en in deze async function?
```javascript
const subcaseType = await subcase.type;
const subcaseTypeWithoutMandatees = [
      CONSTANTS.SUBCASE_TYPES.BEKRACHTIGING,
    ].includes(type?.uri);
```

Ik zou de `get()` op proxy objecten enkel per uitzondering proberen gebruiken, zoals bvb. in simpele getters die niet async kunnen gemaakt worden.

Reply:
we doen net een createRecord
```
const subcase = this.store.createRecord('subcase', {
      type: this.type,
```
Die subcase geven we door aan die method.
Dus de store weet welk type er aanhangt.

Ik kan het aanpassen zodat we van de `get()` af kunnen, maar we kunnen er wel vanuit gaan dat het onmiddelijk resolved?

Reply:
>De includes() gaat niet wachten op het resolven van de promise

Dit doen we ergens anders ook in een getter, en daar wachten we niet echt expliciet om het resolven van de `type`.
Kan dat dan kwaad? Lijkt wel te werken.

Reply:
Changed it, tested it, works

Reply:
> > De includes() gaat niet wachten op het resolven van de promise
> 
> Dit doen we ergens anders ook in een getter, en daar wachten we niet echt expliciet om het resolven van de `type`. Kan dat dan kwaad? Lijkt wel te werken. https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/feature/KAS-3461-remove-mandatees-for-specific-subcase-type/app/controllers/cases/case/subcases/subcase/overview.js#L22

Volgens mij gaat die getter automatisch herberekenen wanneer die promise resolved. En vanaf dan dus de juiste waarde bevatten. Terwijl het hier in een action was en dat niet het geval is.

Je hebt gelijk dat we er nu vanuit kunnen gaan dat die promise onmiddellijk gaat resolven, maar door expliciet te awaiten gaat het in de toekomst ook blijven werken als die assumptie niet meer geldt. Het is een extra lijn code, maar in execution time gaat het geen verschil maken.